### PR TITLE
Pass the pointer by value

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

File: MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs [hunk 1]
```diff
@@ -1519,5 +1519,38 @@ public void ContactsComputeWithPointsThroughAPI()
...
+                    var geometryListIn = new DisposableGeometryList();
+
+                    geometryListIn.XCoordinates = new[] {  1.5, 1.5, 3.5,  3.5,  1.5 };
+                    geometryListIn.YCoordinates = new[] { -1.5, 1.5, 1.5, -1.5, -1.5 };
+                    geometryListIn.Values = new[] { 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 };
+                    geometryListIn.NumberOfCoordinates = geometryListIn.XCoordinates.Length;
+                    geometryListIn.GeometrySeparator = api.GetSeparator();
+
+
+                    int inside = 1;
```
Comment (ahmad-el-sayed) [inline MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs:1544]: can pass 1 to GetSelectedVerticesInPolygon directly?

File: MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs [hunk 2]
```diff
@@ -1519,5 +1519,38 @@ public void ContactsComputeWithPointsThroughAPI()
...
+                    var selectedVertices = api.GetSelectedVerticesInPolygon(id, ref geometryListIn, inside);
+                    Assert.AreEqual(selectedVertices.Length, 4);
+                }
+                finally
+                {
+                    api.DeallocateState(id);
+
```
Comment (ahmad-el-sayed) [inline MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs:1551]: extra space

File: MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs [hunk 3]
```diff
@@ -1519,5 +1519,38 @@ public void ContactsComputeWithPointsThroughAPI()
             }
         }
 
+        [Test]
+        public void GetSelectedVerticesInPolygonThroughAPI()
+        {
+            using (var mesh = GenerateRegularGrid(4, 4, 1, 1))
+            using (var api = new MeshKernelApi())
+            {
+                var id = 0;
```
Comment (ahmad-el-sayed) [inline MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs:1528]: not possible to declare as const int in the try block?

Comment (lucacarniato) [reply MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelTest.cs:1528]: we need the id for later deallocation in the finally block

**Pull request merged**